webrtc-c/canary/src/CloudwatchLogs.cpp

```diff
@@ -0,0 +1,109 @@
+#include "Include.h"
+
+namespace Canary {
+
+CloudwatchLogs::CloudwatchLogs(PConfig pConfig, ClientConfiguration* pClientConfig) : pConfig(pConfig), client(*pClientConfig)
+{
+}
+
+STATUS CloudwatchLogs::init()
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CreateLogGroupRequest createLogGroupRequest;
+    Aws::CloudWatchLogs::Model::CreateLogStreamOutcome createLogStreamOutcome;
+    CreateLogStreamRequest createLogStreamRequest;
+
+    createLogGroupRequest.SetLogGroupName(pConfig->pLogGroupName);
+    // ignore error since if this operation fails, CreateLogStream should fail as well.
+    // There might be some errors that can lead to successfull CreateLogStream, e.g. log group already exists.
+    this->client.CreateLogGroup(createLogGroupRequest);
+
+    createLogStreamRequest.SetLogGroupName(pConfig->pLogGroupName);
+    createLogStreamRequest.SetLogStreamName(pConfig->pLogStreamName);
+    createLogStreamOutcome = this->client.CreateLogStream(createLogStreamRequest);
+
+    if (!createLogStreamOutcome.IsSuccess()) {
+        CHK_ERR(FALSE, STATUS_INTERNAL_ERROR, "Failed to create \"%s\" log stream: %s", pConfig->pLogStreamName,
+                createLogStreamOutcome.GetError().GetMessage().c_str());
+    }
```

Comment:
You could do CHK_ERR(!createLogStreamOutcome.IsSuccess(), STATUS_OPERATION_FAILED, "blah blah);

Reply:
Yep